src/algorithms/libs/galileo_e5_signal_processing.cc

```diff
@@ -123,3 +126,106 @@ void galileo_e5_a_code_gen_complex_sampled(own::span<std::complex<float>> _dest,
+            _dest[index] = std::complex<float>(static_cast<float>(a[0]), 0.0);
+            _dest[index + 1] = std::complex<float>(static_cast<float>(a[1]), 0.0);
+            _dest[index + 2] = std::complex<float>(0.0, 0.0);
+            _dest[index + 3] = std::complex<float>(0.0, 0.0);
+        }
+    else if (_Signal[0] == '5' && _Signal[1] == 'I')
+        {
+            for (size_t i = 0; i < GALILEO_E5B_I_PRIMARY_CODE[prn].length() - 1; i++)
+                {
+                    hex_to_binary_converter(a, GALILEO_E5B_I_PRIMARY_CODE[prn][i]);
+                    _dest[index] = std::complex<float>(static_cast<float>(a[0]), 0.0);
+                    _dest[index + 1] = std::complex<float>(static_cast<float>(a[1]), 0.0);
+                    _dest[index + 2] = std::complex<float>(static_cast<float>(a[2]), 0.0);
+                    _dest[index + 3] = std::complex<float>(static_cast<float>(a[3]), 0.0);
+                    index = index + 4;
+                }
+            // last 2 bits are filled up zeros
+            hex_to_binary_converter(a, GALILEO_E5B_I_PRIMARY_CODE[prn][GALILEO_E5B_I_PRIMARY_CODE[prn].length() - 1]);
+            _dest[index] = std::complex<float>(static_cast<float>(a[0]), 0.0);
+            _dest[index + 1] = std::complex<float>(static_cast<float>(a[1]), 0.0);
+            _dest[index + 2] = std::complex<float>(0.0, 0.0);
+            _dest[index + 3] = std::complex<float>(0.0, 0.0);
+        }
+    else if (_Signal[0] == '5' && _Signal[1] == 'X')
+        {
+            std::array<int32_t, 4> b{};
+            for (size_t i = 0; i < GALILEO_E5B_I_PRIMARY_CODE[prn].length() - 1; i++)
+                {
+                    hex_to_binary_converter(a, GALILEO_E5B_I_PRIMARY_CODE[prn][i]);
+                    hex_to_binary_converter(b, GALILEO_E5B_Q_PRIMARY_CODE[prn][i]);
+                    _dest[index] = std::complex<float>(static_cast<float>(a[0]), static_cast<float>(b[0]));
+                    _dest[index + 1] = std::complex<float>(static_cast<float>(a[1]), static_cast<float>(b[1]));
+                    _dest[index + 2] = std::complex<float>(static_cast<float>(a[2]), static_cast<float>(b[2]));
+                    _dest[index + 3] = std::complex<float>(static_cast<float>(a[3]), static_cast<float>(b[3]));
+                    index = index + 4;
+                }
+            // last 2 bits are filled up zeros
+            hex_to_binary_converter(a, GALILEO_E5B_I_PRIMARY_CODE[prn][GALILEO_E5B_I_PRIMARY_CODE[prn].length() - 1]);
+            hex_to_binary_converter(b, GALILEO_E5B_Q_PRIMARY_CODE[prn][GALILEO_E5B_Q_PRIMARY_CODE[prn].length() - 1]);
+            _dest[index] = std::complex<float>(static_cast<float>(a[0]), static_cast<float>(b[0]));
+            _dest[index + 1] = std::complex<float>(static_cast<float>(a[1]), static_cast<float>(b[1]));
+            _dest[index + 2] = std::complex<float>(0.0, 0.0);
+            _dest[index + 3] = std::complex<float>(0.0, 0.0);
```

Comment:
Maybe these lines should be added to 'void galileo_e5_a_code_gen_complex_primary(own::span<std::complex<float>> _dest,
    int32_t _prn,
    int32_t _prn,
    const std::array<char, 3>& _Signal)
    const std::array<char, 3>& _Signal)'